### PR TITLE
Ammolathe recycling

### LINKED ISSUE #27184: Feature request(?) Ammolathe recycling option, like with the autolathe
#### Description of issue

\>print bullets / magazine
\>shit, wrong bullets / magazine
\>have to throw it in the trash
\>can't stick empty ammo boxes back in either
please fix this, just make it take stuff like the normal autolathe does
would make stocking the damn thing in an emergency a lot easier too

### PR DESCRIPTION
**Pull request by hbycyl**
[powercreep]
closes #27184 

This lets the ammolathe recycle items just like the autolathe does. Also included is a small change to the way the autolathe and its derivatives check whether or not it can recycle an item.

Ammo and guns lose 50% of their material when you recycle them, because fuck security.

:cl:
 * rscadd: The ammolathe can now recycle items

### CODE REVIEW

**Comment by hbycyl** [comment]
I would love to make that warning about accessing fields needing static typing but my understanding is that strings don't get types in byond.

Should I write a getter instead?

**Comment by hbycyl** [comment]
I'm not sure why that failed the map rendering test.

code/game/machinery/autolathe.dm, line 192, column 37:
warning: proc call requires static type: "get_processed_name"

I really have no fucking clue what this is trying to say or how to resolve it.

**Comment by SonixApache** [comment]
technically unatomic but i'll take it
👌

**Comment by DrSnips** [comment]
I can see this _might_ cause some funny hahas but I'm not going to say what those are quite yet because I want to do the funny hahas.

**Comment by hbycyl** [comment]
yes

File: code/modules/projectiles/guns/projectile/rocketlauncher.dm [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	item_state = "rpg"
 	max_shells = 1
 	w_class = W_CLASS_LARGE
-	starting_materials = list(MAT_IRON = 5000)
+	starting_materials = list(MAT_IRON = 25000, MAT_GLASS = 7500, MAT_PLASTIC = 12500, MAT_GOLD = 3000)
```
Comment (jknpj) [inline code/modules/projectiles/guns/projectile/rocketlauncher.dm:9]: what the fuck?

Comment (SonixApache) [reply code/modules/projectiles/guns/projectile/rocketlauncher.dm:9]: yes, those are the actual material costs

**Comment by cloutcode** [comment]
confiscating perp's crowbars to make more vectors

**Pull request merged**